### ISSUE TITLE
Purging variables in pilotCommand installEnv

### ISSUE THREAD

**Issue by andresailer** [DIRACGrid/Pilot#156]
As far as I can tell the GFAL (etc.) variables survive the purging by the diracos2/diracosrc because the original environment is kept from the start
https://github.com/DIRACGrid/Pilot/blob/2a1dc97fc395520a0e19bf0328dba5c9a53dc0c0/Pilot/pilotTools.py#L528
And then only updated with the new values from diracosrc obtained here
https://github.com/DIRACGrid/Pilot/blob/2a1dc97fc395520a0e19bf0328dba5c9a53dc0c0/Pilot/pilotCommands.py#L307
updating installEnv dict
https://github.com/DIRACGrid/Pilot/blob/2a1dc97fc395520a0e19bf0328dba5c9a53dc0c0/Pilot/pilotCommands.py#L316
but non-existing values are not popped
and then the jobagent runs with installEnv
https://github.com/DIRACGrid/Pilot/blob/2a1dc97fc395520a0e19bf0328dba5c9a53dc0c0/Pilot/pilotCommands.py#L1053

Should we just `self.pp.installEnv = {}` before filling updated values?

**Comment by chrisburr** [comment]
> Should we just self.pp.installEnv = {} before filling updated values?

I agree and this is what is done in LHCb already: https://gitlab.cern.ch/lhcb-dirac/LHCbPilot/-/blob/074064f1a1e3b8d4691a42f5e3eeee6ff6dea4ec/LHCbPilot/LHCbPilotCommands.py#L105-117

**Comment by psgebeli** [comment]
Hello @fstagni @andresailer 

It seems like this changed was (accidentally?) removed when the code was refactored in #203 . The current pilot , https://github.com/DIRACGrid/Pilot/blob/master/Pilot/pilotCommands.py , does not clear the install environment, thus potentially harmful variables remain from site environments. 

We (Belle II Distributed Computing) noticed this recently as we are trying to decommission BellePilotCommands in order to use upstream DIRAC pilot commands. However, there are some sites where we are seeing all jobs fail due to these site variables that never get purged.

I can open a pull request, but I wanted to check if this change was intentional? 

Paul (on behalf of Belle II DC)

**Comment by psgebeli** [comment]
As a follow-up, I created a PR: https://github.com/DIRACGrid/Pilot/pull/286
